### ISSUE TITLE
ParamSpec allowed to be used before import

### ISSUE THREAD

**Issue by maharjun** [microsoft/pyright#10799]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

If you have a question about typing or a behavior that you’re seeing in Pyright (as opposed to a bug report or enhancement request), consider posting to the [Pyright discussion forum](https://github.com/microsoft/pyright/discussions).

**Describe the bug**
In the event that the ParamSpec or TypeVar constructs are used prior to being imported from the typing module, they don't report an error if they are imported after their first use.

**Code or Screenshots**

```python
P = ParamSpec("P")  # Should error, doesn't
R = TypeVar("R")

from typing import TypeVar, ParamSpec
```

**Pyright-playground link**

https://pyright-play.net/?code=AoAgvCwIYE5QtgZQA4FMDGAKARMbBKAWACgAlcEAFQE80A1WHUgkkgMxgHt4QAXWgJYA7AOYgB8ZJxi8qtVAxgAaSLAQoMJIA

**version of pyright**
pyright 1.1.403
